src/components/useFetch.ts

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import Todo from "./Todo";
+
+const useFetch = (url: string) => {
+  const [response, setResponse] = useState<Todo[]>([]);
+  const [error, setError]= useState('');
+  const [loader, setLoader]= useState(true);
+  
+  useEffect(() => {
+      const fetchData = async () => {
```

Comment:
You can move this function outside the useEffect. & just call it inside use Effect
```
useEffect(() => {
fetchData()
},[url])
```